### ISSUE TITLE
Change the canonical URL for Guest Posts

### ISSUE THREAD

**Issue by onurgenes** [timlrx/tailwind-nextjs-starter-blog#327]
First of all, thank you for your time to provide such a great template! It is amazing. But I have a question:

**Is your feature request related to a problem? Please describe.**
I want to change canonical URL of the post for guest posting. This surely helps other blogs too. Is there a way to achieve this?

**Describe the solution you'd like**
I can add a new field to the MDX files and change the project accordingly but looking for an easier way.

**Describe alternatives you've considered**
I can change the SEO component depending on my needs but this won't happen all the time but it might be a good option to have.

Thank you!

**Comment by timlrx** [comment]
I think adding it as an optional frontmatter field in the MDX file is probably the easiest and most flexible way. Assuming that you add a field called `canonicalURL`, you would just need to configure the template layout to pass the `canonicalURL` prop to the SEO component and [modify the SEO component](https://github.com/timlrx/tailwind-nextjs-starter-blog/blob/master/components/SEO.js#L140) to use the given canonical URL instead of the default one.

Hope that helps!

**Comment by onurgenes** [comment]
Yes! I was thinking the same thing. Should I open a PR including this?

**Comment by timlrx** [comment]
Yes, that would be great! Would be a pretty convenient feature.

**Comment by onurgenes** [comment]
Sorry for being late but finally starting working on this 🚀

**Comment by onurgenes** [comment]
I have solved this technically but I don't know how to remove blog posts which has `canonicalUrl` in `frontmatter`. Open to ideas to solve this, especially from @timlrx.
